shuup/admin/modules/contact_groups/views/forms.py

```diff
@@ -19,23 +20,36 @@
 
 from shuup.admin.form_part import FormPart, TemplatedFormDef
 from shuup.admin.forms.widgets import ContactChoiceWidget
-from shuup.core.models import Contact, ContactGroup
+from shuup.admin.shop_provider import get_shop
+from shuup.core.models import Contact, ContactGroup, Shop
 from shuup.utils.multilanguage_model_form import MultiLanguageModelForm
 
 
 class ContactGroupBaseForm(MultiLanguageModelForm):
     class Meta:
         model = ContactGroup
-        fields = ("name",)
+        fields = ("name", "shop",)
 
     def __init__(self, *args, **kwargs):
+        self.request = kwargs.pop("request")
         super(ContactGroupBaseForm, self).__init__(*args, **kwargs)
         self.fields['price_display_mode'] = forms.ChoiceField(
             choices=_PRICE_DISPLAY_MODE_CHOICES,
             label=_("Price display mode"),
             initial=_get_price_display_mode(self.instance),
             help_text=_("Set how prices are displayed to contacts in this group."))
 
+        shop = get_shop(self.request)
+        self.fields["shop"] = forms.ModelChoiceField(
+            queryset=Shop.objects.filter(pk=shop.id),
+            initial=shop,
+            widget=HiddenInput(),
+            label=_('shop'),
+            required=False)
+
+    def clean_shop(self):
+        return get_shop(self.request)
```

Comment:
What is this? Consider settings `self.shop = get_shop(self.request)` while you pop out the request at init.

Reply:
It ensures the shop field value never gets set to something we don't want. I'm not sure how the self.shop would help here as it never gets handled.